README.md
# phpadvocat MOVED TO gitlab!
Web application to manage a lawyer's office

This repository is not supported any more 
Code and documentation has been moved to
https://gitlab.com/phpadvocat/phpadvocat/


phpadvocat/CHANGELOG
Version 0.9-4
-----------
2014-01-04
Nikolai Wyderka
- Welcome page is now choosable by the main menue
- new deadline fintions for files, also be shown in calendar and welcome view
- external documents (ODF) can be found in a separate directory
- external documents can be converted in the document view an also be converted in PDF; 
    "lowriter" has to be in PATH

Version 0.9-2
-----------
2013-08-11
Nikolai Wyderka
  Completed calendar view by an week view
  Changed addressee chooser in documents view.
2013-08-29
Burkhard Obergoeker
  changed "similar to" to "like" in search list query in order to be comatible to mysql
  changed query for open invoices to an outer join in order to list also empty invoices 

Version 0.9-1
-----------
2013-06-29
Burkhard Obergoeker
  search now lists also documents
  admin page is able to change the own password (only postgres and mysql))

Version 0.9-0
-----------
2013-06-29
Burkhard Obergoeker
  changed the document part from using LaTeX to FPDF, so
  LaTeX is no longer necessary. The Documents are now
  being stored in the database so for now you got a 
  complete backup when saving the database.

Version 0.8-9
-----------
2013-04-10
Nokolai Wyderka
   added a base number for invoice numbers
   PDF-template for invoice PDF-output now available
   
Burkhard Obergoeker
   added a separate and changable organizational invoice number
   invoice PDF output now in 2 pages
   added "object of service" for invoices

Version 0.8-8
-----------
2013-04-05
Nikolai Wyderka
  added search dialog
  added opposite represantative (lawyer)
  added endnumber to file (german: Ablagenummer)
   
Burkhard Obergoeker
  failed Login jumps back to login page
  added new RVG charges with changed computing and default factor (Germany)
  invoice service period now as field editable


Version 0.8-7
-----------
2013-03-18
Burkhard Obergoeker
add ICS export for events

Version 0.8-6
-----------
2012-11-30
Burkhard Obergoeker
using CSS to give phpadvocat a new look
removing a bug in the calendaredit.php which makes it impossible to
enter events after year 2010

Version 0.8-5
-----------
2012-10-15
Burkhard Obergoeker
removing a bug in the pfiles.php which makes it impossible to
enter events after year 2010

Version 0.8-4
-----------
2009-10-09
Burkhard Obergoeker
Set default Charset of PostgreSQL to LATIN9
eleminated a bug in ms-createdb.sql. Thanks to kereslas!

Version 0.8-3
-----------
2007-01-13
Burkhard Obergoeker
Modified Kontenrahmen for 16% tax in Germany

Version 0.8-2
-----------
2006-01-12
Burkhard Obergoeker
Calender design reconfigured


Version 0.8-1
-----------
2006-01-12
Burkhard Obergoeker
Reconfigured Data model
added accounts and modified categories 


Version 0.8-0
-----------
2005-12-08
Burkhard Obergoeker
Badly dead end for database model
skipped!! (sorry)


Version 0.7-3
-----------
2005-09-14
Burkhard Obergoeker
Changed invoiceedit pay logic to split the expend records into separate
tax percentages. Each record will create a record in the tax bookkeeping
table except the taxrate is 0%.

Version 0.7-2
-----------
2005-07-03
Burkhard Obergoeker
Added online documentation

Version 0.7-1
-----------
Burkhard Obergoeker
Set invoice vat with option list, not checkbox
Optical display total amount with 2 decimals
fixed a major security hole in file access

Version 0.7
-----------
Burkhard Obergoeker
redesigned the bookkeeping module 
changed the data model for bookeeping amounts

Version 0.6
-----------
2005-05-25
Burkhard Obergoeker
added a bookkeeping module 
changed the data model for extended expends
competed the statistics page

Version 0.5-1
-----------
2005-04-22
Burkhard Obergoeker
change data format depending on a global $LOCALE var
correct sql statements to be compatible to MySQL
create a MySQL create script

Version 0.5
-----------
2005-04-15
Burkhard Obergoeker
rename all tables adding prefix "phpa_"
automatism to pay invoices an add the amount to expenditures.
small statistics page
open invoice list

Version 0.4-2
-----------
2005-04-01
Burkhard Obergoeker
redesign of Document (data files) creation
using LaTeX 

Version 0.4-1
-----------
Burkhard Obergoeker
added a MySQL module (not yet tested)
completed pdf-invoice generation
filter for old files in list

Version 0.4
-----------
Burkhard Obergoeker
added pdf output for printing invoices
based on fpdf by Olivier PLATHEY
added a central config table and an 
administration dialog

Version 0.3-1
-----------
2005-02-08
Burkhard Obergoeker
added expenditures
redesign of dialog
display amount of every pfile
create a processregister for every pfile

Version 0.3
-----------
Burkhard Obergoeker
added event manager with calendar function based on
calendrier.php by Mathieu LESNIAK

Version 0.2
-----------
2005-01-22
Burkhard Obergoeker
added invoice manager

Version 0.1
-----------
2005-01-05
Burkhard Obergoeker
- initial version


phpadvocat/calendar.php
<?php
  /**************************************************************************\
  * PHPAdvocat                                                               *
  * http://phpadvocat.sourceforge.net                                        *
  * By Burkhard Obergoeker <phpadvocat@obergoeker.de>                        *
  * --------------------------------------------                             *
  *  This program is free software; you can redistribute it and/or modify it *
  *  under the terms of the GNU General Public License as published by the   *
  *  Free Software Foundation; either version 2 of the License, or (at your  *
  *  option) any later version.                                              *
  \**************************************************************************/

require("./include/phpadvocat.inc.php");
require("./include/calendrier.php");
/* require("./include/phpcalendar.inc.php"); display_month(date("Y"), date("n"));*/

/* Get User Account from Session Vars */ 
$user = $_SESSION["dbuser"];
$passwd = $_SESSION["dbpasswd"];

$changecheck="";

/* initialize database */
$db = new www_db;
$db->connect($user, $passwd);

/****************** add an event -begin- ***************************/
/* add an event */
if($_POST["eventaddbutton"]) {
      /* import POST-VARS */
      $date = $_POST["date"];
      $pnumber = $_POST["pnumber"];
      $eventnumber = $_POST["eventnumber"];
      $start_day = $_POST["start_day"]; /*date=20050219*/
      $start_month = $_POST["start_month"];
      $start_year = $_POST["start_year"];
      $start_hour = $_POST["start_hour"];
      $start_minute = $_POST["start_minute"];
      $description = $_POST["description"];
      $location = $_POST["location"];
      $eventdel = $_POST["eventdel"];

      /* set detail for display */
      $detail=1;
      /* check parameters */
      if(checkdate($start_month, $start_day, $start_year) &&
         ($start_minute < 60) && (start_hour < 24)) {
          
         /* new date variable */
         $date=sprintf("%04.0f%02.0f%02.0f",$start_year, $start_month, $start_day);
          
         /* generate ISO-Format timestamp 1997-12-17 07:37:16-08 */
         $eventstart = sprintf("%s-%s-%s %s:%s:00-00",
          $start_year, $start_month, $start_day, $start_hour, $start_minute);
	   
         /* if eventnumer has not been set, then insert new event, */
         if($eventnumber == 0) {
           $querystring = sprintf("insert into phpa_events ".
             "(pfile, eventstart, description, location) " .
             "values(%s, '%s', '%s', '%s')", 
             $pnumber, $eventstart, $description, $location);
           $changecheck="Termin erstellt";
         } else { /* if number has been set, update event */
           if ($eventdel == "on") {
              $querystring = sprintf("delete from phpa_events ".
                "where number=%s", $eventnumber);
              $changecheck="Termin geloescht";
           } else {
              $querystring = sprintf("update phpa_events set ".
                "pfile=%s, eventstart= '%s', ".
                "description='%s', location='%s' " .
                "where number=%s", 
                $pnumber, $eventstart, $description, $location,
                $eventnumber);
              $changecheck="Termin gespeichert";
           }
         }

         if (!$db->query($querystring)) {
              // $changecheck="Termin erstellt";
         } else {
              $changecheck="Datenbankfehler";
         }
      } else {
            $changecheck="Zeit/Datum nicht korrekt";
      }
}
/****************** add an event -end- ****************************/



echo "<HTML><HEAD><TITLE>PHPAdvocat - Kalender</TITLE>\n";
echo "<meta http-equiv=\"Content-Type\" content=\"text/html; charset=iso-8859-15\">\n";
echo "<link rel=\"stylesheet\" type=\"text/css\" href=\"include/phpadvocat.css\">";
echo "</HEAD>";
echo "<BODY BGCOLOR=\"#FFFFFF\" TEXT=\"#000000\">\n";

/* Begin framework: table with two colunms, */
/* menu on left with suze 200, rest on right */
echo "<TABLE width=100%><TR><TD width=200 valign=top>\n";

  $phpa_menue->account=$_SESSION["dbuser"];
  $phpa_menue->selected = 3;
  $phpa_menue->draw_menue();

echo "</TD><TD>\n"; /* end menue, start dialog */

/* display title */
echo "<CENTER><H1>Kalender</H1></CENTER>\n";

/* display headline */
echo "<table width=100%><tr>\n";
echo "<td>" . date("d.m.Y", time()) . "</td>";
echo "<TD ALIGN=RIGHT></TD>";
echo "</tr></table>\n";

print "<hr><center>";

/* month and day  */
echo "<table width=100% border=0>\n";
echo "<tr>\n";

/* display current month */

/* first evaluate the desired date to display */
// if ($hour == '') {$hour = $_GET['hour']; $date = substr($_GET['hour'],0,8);};
if ($date == '') {$date = $_POST['date'];};
if ($date == '') {$date = $_GET['date'];};
if ($date == '') {$date = date('Ymd',time());};

/* split date string into parts */
$current_month = substr($date, 4 ,2);
$current_day = substr($date, 6, 2);
$current_year = substr($date, 0 ,4);

/* evaluate date for previous and next month */
$previous_month = date("Ymd",     
     mktime( 12,0,0,($current_month - 1),
          01,$current_year));
$next_month = date("Ymd",     
     mktime( 12,0,0,($current_month + 1),
          01,$current_year));

// get events
  $querystring = sprintf("select eventstart ".
       "from phpa_events ".
       "order by eventstart");

  // echo "<hr>" . $querystring . "<hr>";
  $db->query($querystring);

   /* Display all events today */
   $eventarray=array();
   $temparray=array();
   while($db->next_record()) {
       $eventstarttime=
            substr($db->record["eventstart"], 0,4).
            substr($db->record["eventstart"], 5,2).
            substr($db->record["eventstart"], 8,2);
       $eventarray[$eventstarttime] = 1;
   } /* end while */

   // get deadlines
   $querystring = sprintf("select phpa_deadlines.eventday, ".
       "phpa_deadlines.description, phpa_deadlines.type, ".
       "phpa_deadlines.pfile, phpa_pfiles.processregister ".
       "from phpa_deadlines, phpa_pfiles ".
       "where phpa_deadlines.pfile = phpa_pfiles.number ".
       "order by eventday");

   $db->query($querystring);

   /* Display all deadlines today */
   $deadlines=array();
   $predeadlines=array();
   while($db->next_record()) {
       $eventstarttime=
            substr($db->record["eventday"], 0,4).
            substr($db->record["eventday"], 5,2).
            substr($db->record["eventday"], 8,2);
       $eventtext=
           "<a href=\"pfileedit.php?pnumber=" . $db->record["pfile"] . "#detail\">&bull;&nbsp;" . 
           $db->record["description"].'&nbsp;('.
           $db->record["processregister"].')</a><br />';
       if($db->record["type"] == 0) {
           $predeadlines[$eventstarttime] .= $eventtext;
       } else {
           $deadlines[$eventstarttime] .= $eventtext;
       }
   } /* end while */
   
/* set parameters for previous month */
$params['calendar_id'] = 1;
$params['nav_link'] = 1;
$params['short_day_name'] = 0;
$params['lang'] = "german";
$params['use_img']= 0;
$params['link_before_date'] = 1;
$params['link_after_date'] = 1;
$params['font_size'] =12;
//$params['today_bg_color']='white';
$params['highlight'] = $eventarray;
$params['deadlines'] = $deadlines;
$params['predeadlines'] = $predeadlines;

/* display current month */
echo "<td valign=top width=100%><center>\n";
echo calendar($date);
echo "</center></td>\n";

echo "</tr><tr><td><center><a href=$PHP_SELF>Heute</a></center></td></tr>";
/* next line displays all events for the displayed day */
echo "</tr><tr><td valign=top colspan=3><center>\n";


  $querystring = sprintf("select phpa_events.eventstart, ".
       "phpa_events.description, phpa_events.location, ".
       "phpa_events.pfile, phpa_pfiles.processregister ".
       "from phpa_events, phpa_pfiles ".
       "where phpa_events.pfile = phpa_pfiles.number ".
       "order by eventstart");

  // echo "<hr>" . $querystring . "<hr>";
  $db->query($querystring);

   /* Display all events today */
   $eventarray=array();
   $temparray=array();
   while($db->next_record()) {
       $eventstarttime=
            substr($db->record["eventstart"], 0,4).
            substr($db->record["eventstart"], 5,2).
            substr($db->record["eventstart"], 8,2).
            substr($db->record["eventstart"],11,2).
            substr($db->record["eventstart"],14,2);
       $eventtext=
           $db->record["description"].'@'.
           $db->record["location"].' ('.
           $db->record["processregister"].')';
       $eventarray[$eventstarttime] = $eventtext;
   } /* end while */
  
	/* calendar table for current day */
	echo "<hr><table><tr>";
	
	foreach(explode(",", $date) as $day) {
		echo "<td valign=top><center>\n";
		
		/* set parameters for current month */
		$params['calendar_id'] = 2;
		$params['nav_link'] = 1;
		$params['show_day'] = 1;
		$params['show_month'] = 1;
		$params['lang'] = "german";
		$params['use_img']= 0;
		$params['link_before_date'] = 1;
		$params['link_after_date'] = 1;
		$params['today_bg_color']='';
		$params['day_mode'] = 1;
		$params['cell_width'] = '90';
		$params['time_start'] = '6:00';
		$params['time_stop'] = '23:30';
		$params['highlight'] = $eventarray;
		$params['link_on_hour'] = 'calendaredit.php?hour=%%hh%%';
		$params['highlight_type'] = 'text';
		
		/* display current month */
		echo calendar($day);
		echo "</center></td>\n";
		/* end calendar table for current day */
	}

   echo "</tr></table></center></TD></TR></TABLE>\n"; /* end of table framework */

echo "</BODY></HTML>\n"; /* end of page */
?>
